setup-homebrew/main.sh

```diff
@@ -205,7 +208,7 @@ else
         git_retry -C "$HOMEBREW_CORE_REPOSITORY" fetch --force origin
         git -C "$HOMEBREW_CORE_REPOSITORY" remote set-head origin --auto
         git -C "$HOMEBREW_CORE_REPOSITORY" checkout --force -B master origin/HEAD
-    elif [[ -n "${HOMEBREW_NO_INSTALL_FROM_API-}" ]] && [[ ! -d "${HOMEBREW_CORE_REPOSITORY}" ]]; then
+    elif [[ ! -d "${HOMEBREW_CORE_REPOSITORY}" ]]; then
         ohai "Fetching Homebrew/core..."
         git_retry clone https://github.com/Homebrew/homebrew-core "${HOMEBREW_CORE_REPOSITORY}"
```

Comment:
Definitely don't want to always clone Homebrew/core on Linux.

Also worth noting that `HOMEBREW_NO_INSTALL_FROM_API` is not set by default by GitHub on Linux, so some workflows that do https://github.com/Homebrew/brew/blob/master/.github/workflows/tests.yml#L309-L313 will need updating. GitHub's default is a macOS thing.

Reply:
> Definitely don't want to always clone Homebrew/core on Linux.

Yes, messed up this block.

Reply:
@Bo98 Should be fixed!